pallets/claim/src/lib.rs

```diff
@@ -0,0 +1,467 @@
+            beneficiaries: BTreeMap<T::AccountId, BalanceOf<T>>,
+        ) -> DispatchResult {
+            // Check that the pot has enough funds to cover for all the beneficiaries
+            let available_amount = Self::pot();
+            let mut required_amount = TotalClaimable::<T>::get();
+
+            beneficiaries
+                .iter()
+                .try_for_each::<_, DispatchResult>(|(account, amount)| {
+                    if let Some(_) = Beneficiaries::<T>::get(account.clone()) {
+                        // Account already exists
+                        log::warn!("Beneficiary {account:?} already added.");
+                        Err(Error::<T>::AlreadyPresent)?;
+                    } else if amount.is_zero() {
+                        // Attempting to add a beneficiary with nothing to claim
+                        log::warn!("Beneficiary {account:?} with nothing to claim.");
+                        Err(Error::<T>::NothingToClaim)?;
+                    } else {
+                        // Account doesn't exist. Add its token amount to the required amount this pallet's account should have
+                        required_amount = required_amount.saturating_add(*amount);
+                        log::trace!("Added beneficiary {account:?}. Can claim: {amount:?}");
+                    }
+
+                    // Cannot cover for all the tokens, raise an error
+                    if required_amount > available_amount {
+                        Err(Error::<T>::NotEnoughFunds)?;
+                    }
+
+                    Beneficiaries::<T>::insert(account, amount);
+
+                    Ok(())
+                })?;
+
+            // Update total claimable
+            TotalClaimable::<T>::put(required_amount);
+
+            Ok(())
+        }
+
+        fn check_airdrop_status() -> DispatchResult {
+            if !AirdropActive::<T>::get() {
+                Err(Error::<T>::AlreadyEnded)?;
+            }
+            Ok(())
+        }
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Declare the beginning of a new aidrop and start adding beneficiaries (if specified).
+        /// Raise an Error if:
+        /// - There is an already active airdrop
+        /// - There isn't enough balance in the pallets' account to cover for the claim of the supplied beneficiaries (if specified)
+        /// This is an atomic operation. If there isn't enough balance to cover for all the beneficiaries, then none will be added.
+        /// Origin must be the ManagerOrigin.
+        #[pallet::call_index(0)]
+        #[pallet::weight(match &beneficiaries {
+            Some(beneficiaries) => T::WeightInfo::begin_airdrop_with_beneficiaries(beneficiaries.len() as u32),
+            None => T::WeightInfo::begin_airdrop_empty_beneficiaries(),
+        })]
+        pub fn begin_airdrop(
+            origin: OriginFor<T>,
+            beneficiaries: Option<BTreeMap<T::AccountId, BalanceOf<T>>>,
```

Comment:
I'm proposing to remove the option here because I cannot see any advantage on call this function with `None` instead of using an empty map. If you're _scared_ by the `do_add_beneficiaries()` call you can put a guard at the `do_add_beneficiaries()` first line to exit if the map is empty and save two storage read.

This change also should simplify the benchmarks.